src/bindgen/ir/constant.rs

```diff
@@ -174,34 +155,90 @@ impl Literal {
                 ..
             }) => {
                 let struct_name = path.segments[0].ident.to_string();
-                let mut field_pairs: Vec<(String, Literal)> = Vec::new();
+                let mut field_map = HashMap::<String, Literal>::default();
                 for field in fields {
                     let ident = match field.member {
                         syn::Member::Named(ref name) => name.to_string(),
                         syn::Member::Unnamed(ref index) => format!("_{}", index.index),
                     };
                     let key = ident.to_string();
                     let value = Literal::load(&field.expr)?;
-                    field_pairs.push((key, value));
+                    field_map.insert(key, value);
                 }
                 Ok(Literal::Struct {
                     path: Path::new(struct_name.clone()),
                     export_name: struct_name,
-                    fields: field_pairs,
+                    fields: field_map,
                 })
             }
             syn::Expr::Unary(syn::ExprUnary {
                 ref op, ref expr, ..
             }) => match *op {
                 UnOp::Neg(_) => {
                     let val = Self::load(expr)?;
-                    Ok(Literal::Expr(format!("-{}", val)))
+                    Ok(Literal::PostfixUnaryOp {
+                        op: "-",
+                        value: Box::new(val),
+                    })
                 }
                 _ => Err(format!("Unsupported Unary expression. {:?}", *op)),
             },
             _ => Err(format!("Unsupported literal expression. {:?}", *expr)),
         }
     }
+
+    fn write<F: Write>(&self, config: &Config, out: &mut SourceWriter<F>) {
+        match self {
+            Literal::Expr(v) => write!(out, "{}", v),
+            Literal::PostfixUnaryOp { op, ref value } => {
+                write!(out, "{}", op);
+                value.write(config, out);
+            }
+            Literal::BinOp {
+                ref left,
+                op,
+                ref right,
+            } => {
+                left.write(config, out);
+                write!(out, " {} ", op);
+                right.write(config, out);
+            }
+            Literal::Struct {
+                export_name,
+                fields,
+                path,
+            } => {
+                if config.language == Language::C {
+                    write!(out, "({})", export_name);
+                }
+
+                write!(out, "{{ ");
+                let mut is_first_field = true;
+                // In C++, same order as defined is required.
+                let ordered_fields = out
+                    .bindings()
+                    .struct_field_names(path)
+                    .expect("Failed to find struct")
```

Comment:
There's no need to clone() again, nor `expect()` if `struct_field_names` never returns `None`.